assets/js/base/components/cart-checkout/address-form/country-address-fields.js

```diff
@@ -5,542 +5,73 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
+import { __, sprintf } from '@wordpress/i18n';
+import { getSetting } from '@woocommerce/settings';
 
 /**
- * Used to render postcode before the city field.
+ * This is locale data from WooCommerce countries class. This doesn't match the shape of the new field data blocks uses,
+ * but we can import part of it to set which fields are required.
  *
- * @type {Object <AddressFieldKey, AddressField>}
+ * This supports new properties such as optionalLabel which are not used by core (yet).
  */
-const postcodeBeforeCity = {
-	city: {
-		index: 9,
-	},
-	postcode: {
-		index: 7,
-	},
-};
+const coreLocale = getSetting( 'countryLocale', {} );
 
 /**
- * Used to make the state field optional.
+ * Get supported props from the core locale and map to the correct format.
  *
- * @type {Object <AddressFieldKey, AddressField>}
- */
-const optionalState = {
-	state: {
-		required: false,
-	},
-};
-
-/**
- * Used to hide the state field.
+ * Notes:
+ * 	Ignores "class", "type", "placeholder", and "autocomplete"--blocks handles these visual elements.
+ * 	Core "priority" goes up in increments of 10, but blocks expects 1, 2, 3 etc.
  *
- * @type {Object <AddressFieldKey, AddressField>}
+ * @param {Object} localeField Locale fields from WooCommerce.
+ * @return {Object} Supported locale fields.
  */
-const hiddenState = {
-	state: {
-		required: false,
-		hidden: true,
-	},
-};
+const getSupportedProps = ( localeField ) => {
+	const fields = {};
 
-/**
- * Used to hide the postcode field.
- *
- * @type {Object <AddressFieldKey, AddressField>}
- */
-const hiddenPostcode = {
-	postcode: {
-		required: false,
-		hidden: true,
-	},
-};
+	if ( localeField.label !== undefined ) {
+		fields.label = localeField.label;
+	}
```

Comment:
Nitpick: are these if clauses necessary? If `localeField.label` is undefined, I think it's the same assigning it to `fields.label` than not doing it.
```suggestion
	fields.label = localeField.label;
```

Reply:
Correct me if I am wrong, but that would mean `fields` would have a `label ` key, even through it's not needed and the value would be undefined. This could cause breakage later on if this object is spread.

Reply:
Aaah, yeah, good point. Ignore my comment.